repos/repo_result.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2020 Security Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package repos
+
+import (
+	"encoding/csv"
+	"encoding/json"
+	"fmt"
+	"io"
+	"strconv"
+	"strings"
+
+	"github.com/ossf/scorecard/checker"
+)
+
+type RepoResult struct {
+	Repo         string
+	Date         string
+	CheckResults []checker.CheckResult
+	Metadata     []string
+}
+
+func (r *RepoResult) AsJSON(showDetails bool, writer io.Writer) error {
```

Comment:
One more thing: it's not very clear to that this will always output a newline at the end. Would be good to add a document string here for these functions to indicate that's the case.

Reply:
Done.

Reply:
Nit: Per https://golang.org/doc/effective_go#commentary The docstring should start with the name of the function.

> Doc comments work best as complete sentences, which allow a wide variety of automated presentations. The first sentence should be a one-sentence summary that starts with the name being declared.

Reply:
Done.